Add support for getting and setting arbitrary settings via MSPv2

fiam opened the pull request:
Add MSP2_COMMON_SETTING for retrieving arbitrary settings and
MSP2_COMMON_SET_SETTING for setting them. This exposes any setting
which was only available from the CLI via MSP now.

Add support in settings.rb to generate a JSON file with all the
settings, their types and their possible values for settings using
a table. This file can be used by clients to properly format
messages for settings over MSP.

PR for the configurator: https://github.com/iNavFlight/inav-configurator/pull/279

fiam committed:
Add support for getting and setting arbitrary settings via MSPv2

Add MSP2_COMMON_SETTING for retrieving arbitrary settings and
MSP2_COMMON_SET_SETTING for setting them. This exposes any setting
which was only available from the CLI via MSP now.

Add support in settings.rb to generate a JSON file with all the
settings, their types and their possible values for settings using
a table. This file can be used by clients to properly format
messages for settings over MSP.

stronnag commented on src/main/fc/fc_msp.c:
```diff
@@ -1892,6 +1893,148 @@ static mspResult_e mspFcProcessInCommand(uint16_t cmdMSP, sbuf_t *src)
+            break;
+        case VAR_UINT16:
+            FALLTHROUGH;
+            break;
```
Really break here, not just fall through ?

fiam replied:
@stronnag Nice catch!

fiam committed:
Correctly return settings of type VAR_UINT16

Its case was breaking out if switch rather than falling through
to VAR_INT16, which writes the value to the wire.

Kudos to @stronnag for catching it!

fiam committed:
Add missing range check when setting VAR_FLOAT settings

stronnag commented on src/main/fc/fc_msp.c:
```diff
@@ -1892,6 +1893,150 @@ static mspResult_e mspFcProcessInCommand(uint16_t cmdMSP, sbuf_t *src)
+    char name[SETTING_MAX_NAME_LENGTH];
+    uint8_t c;
+    size_t s = 0;
+    while (1) {
```
As MSP will give you the payload size, why not just use that instead of requiring a NUL terminating byte (the consumer has to supply the payload size regardless).

stronnag replied:
Huum, I'd not considered the set use case, where the sum of name and value is somewhat less determinate.

fiam replied:
@stronnag  that’s right, you’re in a catch 22 without an explicit terminator or name length. You need to know the setting name in order to find how big its data is. The getter could do without the zero terminator, but I thought using the same encoding for getter and setter would be more intuitive. I’d expect these new commands to be used mostly for configuration and not for live updates, so I thought the extra byte would not make a difference.

fiam committed:
Add setting_get_value_size() function and use it for MSP

Used to reply to MSP2_COMMON_SETTING, since we're just copying
the value from memory to the wire. Will be used in more places
eventually.
